src/sagemaker_xgboost_container/data_utils.py

```diff
@@ -299,22 +452,37 @@ def get_dmatrix(data_path, content_type, csv_weights=0):
                         1 if the instance weights are in the second column of csv file; otherwise, 0
     :return: xgb.DMatrix
     """
-    if not os.path.exists(data_path):
+    if not (os.path.exists(data_path) or (is_pipe and os.path.exists(data_path + '_0'))):
         return None
     else:
         if os.path.isfile(data_path):
             files_path = data_path
-        else:
+        elif not is_pipe:
             for root, dirs, files in os.walk(data_path):
                 if dirs == []:
                     files_path = root
                     break
         if content_type.lower() == CSV:
-            dmatrix = get_csv_dmatrix(files_path, csv_weights)
+            if is_pipe:
```

Comment:
Ideally this logic would look something like

reader = get_content_reader('csv') // this would return the correct reader for the format for example CSVReader for csv

dmatrix = get_dmatrix(reader, args, pipe)

I know that that requires changing code out of cope for this task, but can we at least add a todo?

Reply:
There would need to be a few prerequisite changes in order for this to work:
- Use mlio for csv file mode 
- Have a data reader for libsvm (not yet supported)
- Have a data reader for parquet (the current ParquetRecordReader is actually an iterator over the records and does not follow the same interface as the other readers)

Should I leave a todo as a comment?